dart/lib/src/client_reports/client_report_recorder.dart

```diff
@@ -0,0 +1,53 @@
+import 'package:meta/meta.dart';
+
+import '../utils.dart';
+import 'client_report.dart';
+import 'discarded_event.dart';
+import 'outcome.dart';
+import '../transport/rate_limit_category.dart';
+
+@internal
+class ClientReportRecorder {
+  ClientReportRecorder([this._dateTimeProvider = getUtcDateTime]);
+
+  final DateTime Function() _dateTimeProvider;
+  final Map<_QuantityKey, int> _quantities = {};
+
+  void recordLostEvent(final Outcome reason, final RateLimitCategory category) {
```

Comment:
We need to double check if `RateLimitCategory` is accurate and if it's missing new entries.
See https://github.com/getsentry/develop/pull/535/files and https://github.com/getsentry/develop/pull/536/files

Reply:
Hmm, hard to say. We are missing the `internal` type  in `RateLimitCategory`, but have the `__all__` type, which i think is specific to the rate limiting implementation and should probably never be used here. Should we create a separate enum type to be on the safe side?

Reply:
Let's wait those 2 PRs get reviewed/merged, we can rename to `DataCategory` at least in case this is not a breaking change to public APIs

Reply:
Yes, I also renamed `RateLimitCategory` to `DataCategory` on Cocoa as relay uses `DataCategory` for both rate limits and client reports.

Reply:
Can you point me to the appropriate PR? How are you candling the "all" case, assuming that it should not be valid in the client report context and is specific to to rate limiting?

Reply:
Just wondering, as the Dart implementation was ported from `sentry-java` and we have an `all` category there: https://github.com/getsentry/sentry-java/blob/b45944c93d53f68bc1e64f5ac4c9a5400b252d44/sentry/src/main/java/io/sentry/transport/RateLimiter.java#L25